orangecontrib/prototypes/widgets/owinteractions.py

```diff
@@ -0,0 +1,242 @@
+"""
+Interactions widget
+"""
+from operator import attrgetter
+from itertools import chain
+
+import numpy as np
+
+from AnyQt.QtCore import Qt
+from AnyQt.QtGui import QStandardItem
+
+from Orange.data import Table, Domain, ContinuousVariable, StringVariable
+from Orange.widgets import gui
+from Orange.widgets.utils.itemmodels import DomainModel
+from Orange.widgets.utils.signals import Input, Output
+from Orange.widgets.utils.widgetpreview import WidgetPreview
+from Orange.widgets.widget import OWWidget, AttributeList, Msg
+from Orange.preprocess import Discretize, Remove
+from Orange.preprocess.discretize import EqualFreq
+import Orange.widgets.data.owcorrelations
+
+
+SIZE_LIMIT = 1000000
+
+
+class Interaction:
```

Comment:
The class `Interaction` should be tested separately, independently of widget tests, in its own TestSomething class. The same for heuristics.

Reply:
What this tests should test is what your results should be. They should also test behavior in strange cases, like unknown values (if the class supports them).

Reply:
I've moved the tests to their own classes and added tests for sparse data.